src/optimal_planner.cpp

```diff
@@ -1249,16 +1249,17 @@ bool TebOptimalPlanner::isHorizonReductionAppropriate(const std::vector<geometry
     return false;
   
   // check if distance is at least 2m long // hardcoded for now
-  double dist = 0;
-  for (int i=1; i < teb_.sizePoses(); ++i)
   {
```

Comment:
Thanks for your suggested changes. We could also resolve shadowing by replacing `double dist = ` by `dist = ` in line 1299. Therefore, no separate separate scope would have to be added. What do you think?

Reply:
Yes, or even better we should rename the variable. This would ensure that if in the future any part of this code was restructured or moved around the `dist` variable would not be altered in an uexpected way.

Just let me know your preference and I'll edit it.

Reply:
your right, I am fine with renaming the variable! Thanks a lot!